IntegrationWithPharmacies/Controllers/SharingPrescriptionController.cs

```diff
@@ -1,30 +1,198 @@
+
+        private IActionResult GetMedicineAvailabilityTable(string availability)
+        {
+            List<MedicineName> medicines = new List<MedicineName>();
+            String[] fileParts = availability.Split(";");
+            if (fileParts.Length == 0) GetOnlyOnePharmacy(availability, medicines);
+            else GetAllPharmacies(medicines, fileParts);
+            return Ok(medicines);
+        }
+
+        private static void GetAllPharmacies(List<MedicineName> medicines, string[] fileParts)
+        {
+            for (int i = 0; i < fileParts.Length; i++)
+            {
+                String[] nameParts = fileParts[i].Split("_");
+                medicines.Add(new MedicineName("Pharmacy: " + nameParts[0] + ", city: " + nameParts[1], nameParts[2]));
+            }
+        }
+
+        private static void GetOnlyOnePharmacy(string availability, List<MedicineName> medicines)
+        {
+            String[] nameParts = availability.Split("_");
+            MedicineName medicine = new MedicineName("Pharmacy: " + nameParts[0] + ", city: " + nameParts[1], nameParts[2]);
+            medicines.Add(medicine);
+        }
+
+        private int getRandomNumber()
+        {
+            return new Random().Next(1, 100);
+        }
+        private String getTextForPrescription(Prescription prescription)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            return stringBuilder.Append(prescription.Pharmacy+"          Precription for medicine\n\nPatients name: " + prescription.Name + "\nPatients surname: " + prescription.Surname + "\nPatients medical ID number: " + prescription.MedicalIDNumber + "\nMedication: " + prescription.Medicine + "     Quantity: " + prescription.Quantity + "\nUsage: " + prescription.Usage + "\n").ToString();
+         
+        }
+        [HttpPost("http")]
+        public IActionResult PostHttp(Prescription prescription)
+        {
+            if (Environment.Equals("Development"))return sendPrescriptionHttp(prescription);
+            return BadRequest();
+        }
+
+        private IActionResult sendPrescriptionHttp(Prescription prescription)
+        {
+            String complete = @"FilePrescriptions\..\Prescription" + DateTime.Now.ToString("dd-MM-yyyy") + "_" + getRandomNumber() + ".txt";
+            System.IO.FileStream fs = System.IO.File.Create(complete);
+            fs.Close();
+            System.IO.File.WriteAllText(complete, getTextForPrescription(prescription));
+            try
+            {   uploadFile(complete);
+                return Ok();
+            }
+            catch (Exception e) { return BadRequest(); }
+           
+        }
+
+        public void uploadFile(String complete)
+        {
+            WebClient client = new WebClient();
+            Uri uri = new Uri(@"http://localhost:8082/download/prescription/http");
+            client.Credentials = CredentialCache.DefaultCredentials;
+            client.UploadFile(uri, "POST", complete);

```

Comment:
Apply a constructor with parameters for the WebClient, and pass the 'uri' object directly to the 'UploadFile ()' method.